app/services/hotkeys.ts

```diff
@@ -240,166 +284,179 @@ export class HotkeysService extends StatefulService<IHotkeysServiceState> {
+
+      if (!addedHotkeys.has(key)) {
         hotkeys.push({
+          [idPropFor(hotkey)]: hotkey.ObjectName,
           actionName: action.name,
-          bindings: [],
-          sourceId: source.sourceId
+          bindings: [] as IBinding[],
+          hotkeyId: hotkey.HotkeyId,
         });
-      });
+        addedHotkeys.add(key);
+      }
     });
 
     // Set up bindings from saved hotkeys
     // This is a slow O(n^2) process, and may need to
     // be optimized later.
     this.state.hotkeys.forEach(savedHotkey => {
       const hotkey = hotkeys.find(blankHotkey => {
-        return this.getHotkey(blankHotkey).isSameHotkey(savedHotkey);
+        return this.getHotkey(blankHotkey).equals(savedHotkey);
       });
-      if (hotkey) hotkey.bindings = [].concat(savedHotkey.bindings);
+      if (hotkey) {
+        hotkey.bindings = [].concat(savedHotkey.bindings);
+      }
     });
 
-    this.registeredHotkeys = hotkeys.map(hotkeyModel => this.getHotkey(hotkeyModel));
+    this.registeredHotkeys = hotkeys.map(this.getHotkey);
   }
 
-
   getHotkey(obj: IHotkey): Hotkey {
     return new Hotkey(obj);
   }
 
-
   getHotkeys(): Hotkey[] {
-    if (!this.registeredHotkeys) this.updateRegisteredHotkeys();
+    if (!this.registeredHotkeys) {
+      this.updateRegisteredHotkeys();
+    }
     return this.registeredHotkeys.filter(hotkey => hotkey.shouldApply);
   }
 
-
   getHotkeysSet(): IHotkeysSet {
-
     const sourcesHotkeys: Dictionary<Hotkey[]> = {};
     this.sourcesService.getSources().forEach(source => {
       const sourceHotkeys = this.getSourceHotkeys(source.sourceId);
-      if (sourceHotkeys.length) sourcesHotkeys[source.sourceId] = sourceHotkeys;
+      if (sourceHotkeys.length) {
+        sourcesHotkeys[source.sourceId] = sourceHotkeys;
+      }
     });
 
     const scenesHotkeys: Dictionary<Hotkey[]> = {};
     this.scenesService.scenes.forEach(scene => {
       const sceneItemsHotkeys = this.getSceneItemsHotkeys(scene.id);
       const sceneHotkeys = sceneItemsHotkeys.concat(this.getSceneHotkeys(scene.id));
-      if (sceneHotkeys.length) scenesHotkeys[scene.id] = sceneHotkeys;
+      if (sceneHotkeys.length) {
+        scenesHotkeys[scene.id] = sceneHotkeys;
+      }
     });
 
     return {
       general: this.getGeneralHotkeys(),
       sources: sourcesHotkeys,
-      scenes: scenesHotkeys
+      scenes: scenesHotkeys,
     };
   }
 
-
   clearAllHotkeys() {
     this.applyHotkeySet({
       general: [],
       sources: {},
-      scenes: {}
+      scenes: {},
     });
   }
 
-
   applyHotkeySet(hotkeySet: IHotkeysSet) {
     const hotkeys: IHotkey[] = [];
     hotkeys.push(...hotkeySet.general);
     Object.keys(hotkeySet.scenes).forEach(sceneId => hotkeys.push(...hotkeySet.scenes[sceneId]));
-    Object.keys(hotkeySet.sources).forEach(sourceId => hotkeys.push(...hotkeySet.sources[sourceId]));
+    Object.keys(hotkeySet.sources).forEach(sourceId =>
+      hotkeys.push(...hotkeySet.sources[sourceId]),
+    );
```

Comment:
are these all automatic linter changes? i feel like we should change the config because we use single line ifs and anonymous functions all the time

Reply:
For style issues, I believe we should focus our effort on #1129. I believe it'd be wasteful to change something back to just change it again (or not) in the other branch. This might've been before we merged #1043, so I apologize if it was a result of my editor and switching branches.